### PR TITLE
Allow specific origin on OPTIONS requests.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jrwren** [comment]
:+1:

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1131/
Test PASSed.

File: internal/router/router.go [hunk 1]
```diff
@@ -268,6 +269,7 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		// putting OPTIONS handling in every endpoint,
 		// and it shouldn't actually matter in practice.
 		header.Set("Allow", "DELETE,GET,HEAD,PUT,POST")
+		header.Set("Access-Control-Allow-Origin", req.Header.Get("Origin"))
```
Comment (rogpeppe) [inline internal/router/router.go:272]: Shouldn't we only do this if the Origin header is actually set?

**Pull request merged**